### PR TITLE
CSS cleanup (pre Stylelint)

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

This PR is a preliminary application of Stylelint rules to the current codebase, to reduce the surface area of https://github.com/hashicorp/design-system/pull/543 (and the discussion around rules to activate).

The changes are mainly fixes for formatting issues, wrong code case, existing patterns not followed in some cases, etc (nothing controversial) so this PR should be pretty straightforward. If you think some rules/changes should be discussed before applying them, add a comment in the PR and I can remove the commit if we agree to discuss it later more in detail.

### :hammer_and_wrench: Detailed description

In this PR I have made these changes:
- indentation (from `4` to `2` spaces like all the other files)
- extra semicolon ([`no-extra-semicolons`](https://stylelint.io/user-guide/rules/list/no-extra-semicolons/)) 
- extra space in var() declarations ([`function-parentheses-space-inside`](https://stylelint.io/user-guide/rules/list/function-parentheses-space-inside/))
- removed unit from zero values ([`length-zero-no-unit`](https://stylelint.io/user-guide/rules/list/length-zero-no-unit/))
- removed extra spaces ([`no-eol-whitespace`](https://stylelint.io/user-guide/rules/list/no-eol-whitespace/))
- added missing EOL ([`no-missing-end-of-source-newline`](https://stylelint.io/user-guide/rules/list/no-missing-end-of-source-newline/))
- fixed hex color value case ([`color-hex-case`](https://stylelint.io/user-guide/rules/list/color-hex-case/))
- remove unnecessary vendor-prefixed declaration ([`property-no-vendor-prefix`](https://stylelint.io/user-guide/rules/list/property-no-vendor-prefix/))
- fixed whitespace in comment ([`scss/double-slash-comment-whitespace-inside`](https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/double-slash-comment-whitespace-inside/README.md))
- fixed selector ([`selector-combinator-space-after`](https://stylelint.io/user-guide/rules/list/selector-combinator-space-after/))
- remove extra newlines
- fixed double space ([`declaration-colon-space-after`](https://stylelint.io/user-guide/rules/list/declaration-colon-space-after/))
- fixed a couple of spacing issues ([`at-rule-empty-line-before`](https://stylelint.io/user-guide/rules/list/at-rule-empty-line-before/))
- fixed some spacing ([`scss/dollar-variable-empty-line-before`](https://github.com/stylelint-scss/stylelint-scss/blob/master/src/rules/dollar-variable-empty-line-before/README.md))

### :link: External links

- [Epic in Jira](https://hashicorp.atlassian.net/browse/HDS-496)

***

### 👀 How to review

👉 Review commit-by-commit (suggested, if you want to see the individual changes according to stylelint rules) 
👉 Review by files changed (if you want to see the changes overall).

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by alex-ju — APPROVED** [review]
All looks really good to me! The only one that we might need to disable from time to time would be the vendor-prefixed declaration.

File: packages/components/app/styles/components/icon-tile.scss [hunk 1]
```diff
@@ -58,13 +57,13 @@ $size-props: (
...
-      'extra-size': 1.5rem, // 24px
-      'extra-border-radius': 5px,
-      'extra-icon-size': 1rem,
+    "size": 2.5rem, // 40px
+    "border-radius": 6px,
+    "icon-size": 1.5rem, // 24px
+    "logo-size": 1.75rem, // 28px
+    'extra-size': 1.5rem, // 24px
+    'extra-border-radius': 5px,
+    'extra-icon-size': 1rem,
```
Comment (KristinLBradley) [inline packages/components/app/styles/components/icon-tile.scss:66]: I'd like to see a consistent quote style. 

(My personal preference has been to use double quotes within HTML & CSS and single quotes within JS.)

**Comment by KristinLBradley** [comment]
> All looks really good to me! The only one that we might need to disable from time to time would be the vendor-prefixed declaration.

I agree. The rules all look good but there are occasional exceptions requiring vendor-prefixes. But if we can just disable those where needed that seems fine to disallow as a general rule.

File: packages/components/app/styles/components/icon-tile.scss (hunk 1 above)
Comment (didoo) [reply packages/components/app/styles/components/icon-tile.scss:66]: Yes, this will be discussed in https://github.com/hashicorp/design-system/pull/543

**Comment by didoo** [comment]
To avoid too many conflicts, since we already have two approvals, I am going to merge this PR. 

@MelSumner if you think some of the rules used applied above need more discussion, we can do it in https://github.com/hashicorp/design-system/pull/543 (where there will be a wider discussion about the rules declarations).

**Comment by MelSumner** [comment]
@didoo I think this is great, excited to try it out.

**Pull request merged**